### PR TITLE
created DTO,mapper,exceptionClass,controller.

### PR DESCRIPTION
**Pull request by opopovych**
all classes are implemented and was add missing method to previous task

### CODE REVIEW

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+
```
Comment (koval-kn) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java]: it's decided by corporative standarts either to put blank lines here and at the end of the class or not, but it definetely should be in the same style through all the project - so choose one style and apply it

File: src/main/java/mate/academy/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @GetMapping("get/books/{id}")
+    public BookDto getById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    @PostMapping("post/books")
+    public BookDto createBook(@RequestBody CreateBookRequestDto createBookRequestDto) {
+        return bookService.save(createBookRequestDto);
+    }
+
```
Comment (koval-kn) [inline src/main/java/mate/academy/controller/BookController.java]: redundant blank line

File: src/main/java/mate/academy/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class BookController {
+    @Autowired
+    private BookService bookService;
+
+    @GetMapping("get/books")
```
Comment (koval-kn) [inline src/main/java/mate/academy/controller/BookController.java]: invalid url mapping
image

File: src/main/java/mate/academy/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("get/books/{id}")
```
Comment (koval-kn) [inline src/main/java/mate/academy/controller/BookController.java]: invalid url mapping

File: src/main/java/mate/academy/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    public BookDto getById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    @PostMapping("post/books")
```
Comment (koval-kn) [inline src/main/java/mate/academy/controller/BookController.java]: invalid url mapping

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/controller/BookController.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import mate.academy.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class BookController {
```
Comment (sokimaaa) [inline src/main/java/mate/academy/controller/BookController.java:17]: you have the same parts of URI that may be extracted by using @RequestMapping on a class

**Pull request merged**